torch/_meta_registrations.py

```diff
@@ -18,6 +18,57 @@ def toRealValueType(dtype):
     return from_complex.get(dtype, dtype)
 
 
+def toComplexValueType(dtype):
+    to_complex = {
+        torch.half: torch.complex32,
+        torch.float: torch.cfloat,
+        torch.double: torch.cdouble,
+    }
+    return to_complex[dtype]
+
+
+@torch.library.impl(meta_lib, "_fft_c2c")
+def meta_fft_c2c(self, dim, normalization, forward):
+    assert self.dtype.is_complex
+    return self.new_empty(list(self.size()))
+
+
+@torch.library.impl(meta_lib, "_fft_r2c")
+def meta_fft_r2c(self, dim, normalization, onesided):
+    assert self.dtype.is_floating_point
+    output_sizes = list(self.size())
+
+    if onesided:
+        last_dim = dim[-1]
+        last_dim_halfsize = int((output_sizes[last_dim] / 2) + 1)
```

Comment:
Instead of casting to int after the floating point computation, why not do integer division. This will make it easier to symbolically trace this as we won't go into floats temporarily here